lib/internal/modules/esm/default_resolve.js

```diff
@@ -52,8 +53,9 @@ if (experimentalJsonModules)
   extensionFormatMap['.json'] = legacyExtensionFormatMap['.json'] = 'json';
 
 function resolve(specifier, parentURL) {
+  let parsed;
   try {
-    const parsed = new URL(specifier);
```

Comment:
why did you move this declaration?

Reply:
We cannot throw from the try block, because the catch block is empty and the error will be lost. And if we want to access the `parsed` outside the try..catch block it needs to be declared outside. Or do you have something other in mind?